runtime/js/40_command.js

```diff
@@ -0,0 +1,198 @@
+      stdout,
+      stderr,
+    });
+    return new Child(illegalConstructorKey, child);
+  }
+
+  class Child {
+    #rid;
+
+    #pid;
+    get pid() {
+      return this.#pid;
+    }
+
+    #stdinRid;
+    #stdin = null;
+    get stdin() {
+      return this.#stdin;
+    }
+
+    #stdoutRid;
+    #stdout = null;
+    get stdout() {
+      return this.#stdout;
+    }
+
+    #stderrRid;
+    #stderr = null;
+    get stderr() {
+      return this.#stderr;
+    }
+
+    constructor(key = null, {
+      rid,
+      pid,
+      stdinRid,
+      stdoutRid,
+      stderrRid,
+    } = null) {
+      if (key !== illegalConstructorKey) {
+        throw new TypeError("Illegal constructor.");
+      }
+
+      this.#rid = rid;
+      this.#pid = pid;
+
+      if (stdinRid !== null) {
+        this.#stdinRid = stdinRid;
+        this.#stdin = writableStreamForRid(stdinRid);
+      }
+
+      if (stdoutRid !== null) {
+        this.#stdoutRid = stdoutRid;
+        this.#stdout = readableStreamForRid(stdoutRid);
+      }
+
+      if (stderrRid !== null) {
+        this.#stderrRid = stderrRid;
+        this.#stderr = readableStreamForRid(stderrRid);
+      }
+    }
+
+    #status;
+    get status() {
+      if (this.#status) {
+        return this.#status;
+      }
+      const status = core.opSync("op_command_status", this.#rid);
+      // TODO(@crowlKats): 2.0 change typings to return null instead of undefined for status.signal
+      if (status) {
+        status.signal ??= undefined;
+      }
+      return status;
+    }
+
+    async wait() {
+      if (this.#rid === null) {
+        throw new TypeError("Child process has already terminated.");
+      }
+      const status = await core.opAsync(
+        "op_command_wait",
+        this.#rid,
+        this.#stdinRid,
+      );
+      await this.stdin?.abort();
+      this.#rid = null;
+      // TODO(@crowlKats): 2.0 change typings to return null instead of undefined for status.signal
+      status.signal ??= undefined;
+      this.#status = status;
+      return status;
+    }
+
+    async output() {
+      if (this.#rid === null) {
+        throw new TypeError("Child process has already terminated.");
+      }
+      const res = await core.opAsync("op_command_output", {
+        rid: this.#rid,
+        stdoutRid: this.#stdoutRid,
+        stderrRid: this.#stderrRid,
+      });
+      await this.stdin?.abort();
+      this.#rid = null;
+      // TODO(@crowlKats): 2.0 change typings to return null instead of undefined for status.signal
+      res.status.signal ??= undefined;
+      this.#status = res.status;
+      return res;
+    }
+
+    kill(signo) {
+      if (this.#rid === null) {
+        throw new TypeError("Child process has already terminated.");
+      }
+      core.opSync("op_kill", this.#pid, signo);
+    }
+  }
+
+  function command(command, {
+    stdin = "null",
+    stdout = "piped",
+    stderr = "piped",
+    ...options
+  } = {}) { // TODO(@crowlKats): more options (like input)?
```

Comment:
By input you mean a string that should be written into the stdin? If so I think we should add it. Also it's a bit strange that one can configure `stdin` if there's no way to write to it.

Reply:
can we actually keep that for a follow-up? would simplify things

Reply:
Yes, please open an issue about it before landing so we don't forget about it